src/main/main.cpp

```diff
@@ -138,10 +127,8 @@ auto main(int argc, char *argv[]) -> int {
 
   /* If there are no parser errors then proceed to generate the parse tree */
   if (err.empty()) {
-    std::ofstream ofs(output_file, std::ofstream::out);
     parser.BuildParseTree();
-    ofs << jucc::parser::Parser::FormattedJSON(parser.GetParseTree()).dump() << '\n';
-    ofs.close();
+    parser.WriteParseTree(output_file, true);
```

Comment:
Add error handling, if error `WriteParseTree` will return false.

`true` i.e formatted is default argument, not required.

Reply:
what😂

Could you please recheck after updating the signature in header file. Hope it should resolve the issue

Reply:
On changing headers
```
CMakeFiles/jucc_objlib.dir/src/parser/parser.cpp.o
/Library/Developer/CommandLineTools/usr/bin/c++ -DJSON_DIAGNOSTICS=0 -DJSON_USE_IMPLICIT_CONVERSIONS=1 -DNDEBUG -I../src/include -Inlohmann-json-src/single_include -O3 -DNDEBUG -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX11.1.sdk -fPIC -Werror -Wall -march=native -mcx16 -O3 -std=c++17 -MD -MT CMakeFiles/jucc_objlib.dir/src/parser/parser.cpp.o -MF CMakeFiles/jucc_objlib.dir/src/parser/parser.cpp.o.d -o CMakeFiles/jucc_objlib.dir/src/parser/parser.cpp.o -c ../src/parser/parser.cpp
../src/parser/parser.cpp:163:63: error: redefinition of default argument
bool Parser::WriteParseTree(const std::string &filepath, bool formatted = true) {
                                                              ^           ~~~~
../src/include/parser/parser.h:127:57: note: previous definition is here
  bool WriteParseTree(const std::string &filepath, bool formatted=true);
                                                        ^         ~~~~
1 error generated.
```

Without changing headers

```
 CMakeFiles/jucc.dir/src/main/main.cpp.o
/Library/Developer/CommandLineTools/usr/bin/c++ -DJSON_DIAGNOSTICS=0 -DJSON_USE_IMPLICIT_CONVERSIONS=1 -DNDEBUG -I../src/include -Inlohmann-json-src/single_include -O3 -DNDEBUG -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX11.1.sdk -Werror -Wall -fvisibility=hidden -march=native -mcx16 -O3 -std=gnu++17 -MD -MT CMakeFiles/jucc.dir/src/main/main.cpp.o -MF CMakeFiles/jucc.dir/src/main/main.cpp.o.d -o CMakeFiles/jucc.dir/src/main/main.cpp.o -c ../src/main/main.cpp
../src/main/main.cpp:131:43: error: too few arguments to function call, expected 2, have 1
    if (!parser.WriteParseTree(output_file)) {
         ~~~~~~~~~~~~~~~~~~~~~            ^
../src/include/parser/parser.h:127:3: note: 'WriteParseTree' declared here
  bool WriteParseTree(const std::string &filepath, bool formatted);
  ^
1 error generated.
```

Reply:
keep true in parser.h and remove it from parser.cpp

Reply:
default arg not required in cpp file only required in h file

Reply:
Yes it worked.